lib/sqlalchemy/ext/mypy/decl_class.py

```diff
@@ -462,25 +483,18 @@ def _scan_for_mapped_bases(
 
     """
 
-    info = util._info_for_cls(cls, api)
+    info = util.info_for_cls(cls, api)
 
-    baseclasses = list(info.bases)
-
-    while baseclasses:
-        base: Instance = baseclasses.pop(0)
+    if info is None:
+        return
 
```

Comment:
**mike bayer** (zzzeek) wrote:

this seems to be where the biggest conceptual change is actually made; removing the "mapped_mro" collection and evaluating it later when needed instead.   guess this makes the scan for "mapped mro" fresher since you do it each time.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2962